orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/SimpleTask.java

```
@@ -0,0 +1,99 @@
+import javax.annotation.Nullable;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.core.GenericTypeResolver;
+import org.springframework.core.ResolvableType;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+public class SimpleTask implements Task {
+  private SimpleStage simpleStage;
+
+  SimpleTask(@Nullable SimpleStage simpleStage) {
+    this.simpleStage = simpleStage;
+  }
+
+  private SimpleStageInput getStageInput(Stage stage) {
+    ObjectMapper objectMapper = OrcaObjectMapper.newInstance();
+
+    try {
+      List<Class<?>> cArg = Arrays.asList(SimpleStageInput.class);
+      Method method = simpleStage.getClass().getMethod("execute", cArg.toArray(new Class[0]));
+      Type inputType = ResolvableType.forMethodParameter(method, 0).getGeneric().getType();
+      Map<TypeVariable, Type> typeVariableMap =
+          GenericTypeResolver.getTypeVariableMap(simpleStage.getClass());
+
+      return new SimpleStageInput(
+          objectMapper.convertValue(
+              stage.getContext(), GenericTypeResolver.resolveType(inputType, typeVariableMap)));
+    } catch (NoSuchMethodException exeception) {
+      log.error("Cannot get method for " + simpleStage.getName() + ": " + exeception.getMessage());
+    }
+
+    return new SimpleStageInput(new Object());
+  }
+
+  @Nonnull
+  public TaskResult execute(@Nonnull Stage stage) {
+    ExecutionStatus status;
+    SimpleStageInput simpleStageInput = getStageInput(stage);
+    SimpleStageOutput output = simpleStage.execute(simpleStageInput);
+
+    switch (output.getStatus()) {
+      case TERMINAL:
+        status = ExecutionStatus.TERMINAL;
+        break;
+      case RUNNING:
+        status = ExecutionStatus.RUNNING;
+        break;
+      case COMPLETED:
+        status = ExecutionStatus.SUCCEEDED;
+        break;
+      case NOT_STARTED:
+        status = ExecutionStatus.NOT_STARTED;
+        break;
+      default:
+        status = ExecutionStatus.FAILED_CONTINUE;
```

Should we just raise an exception on an unknown status ... `FAILED_CONTINUE` doesn't seem like the appropriate default.